lib/honeybadger/worker.rb

```diff
@@ -24,7 +24,7 @@ def initialize(max_size)
       end
 
       def push(msg)
-        super unless size == max_size
+        super unless size >= max_size
```

Comment:
I think this is an acceptable fix. Would we also want to synchronize the size check to avoid the concurrent write issue entirely? I think this was an oversight when rebuilding this part of the library--this really should be thread safe like `Queue` is by default, unless I'm missing something.

Reply:
I think Ruby's `Queue` already has a mutex available, although it might be janky to rely on it since it appears to be an environment variable. What do you think is the best solution here, @minhajuddin?

Reply:
@joshuap Here is an alternative proposal if you want to switch this over to the ruby stdlib queue. (I believe this would require ruby 2.2, not sure if that is a problem)

```ruby
q = SizedQueue.new(1)
q.push("hello", true) 
begin
  q.push("hello", true) # non_block set to true, error is raised if queue is full
rescue ThreadError 
  nil
end
```

Reply:
We still need to support Ruby 2.1, and I'd rather not use exception handling for this because it's a little less performant. I'd rather add our own mutex to `Honeybadger::Queue` and then do something like this in `Queue#push`:

```ruby
      def push(msg)
        @our_mutex.synchronize do
          super unless size >= max_size
        end
      end
```